File: sorting/squares-of-a-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def sorted_squares(nums)
    res = []
    i, j = 0, nums.size-1
    while i <= j
        if nums[i]**2 > nums[j]**2
            res << nums[i]**2
            i += 1
        else
            res << nums[j]**2
            j -= 1
        end
    end
    res.reverse
end

File: two-pointers/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = []
        left, right = 0, len(nums)-1
        while left <= right:
            l_square = nums[left] * nums[left]
            r_square = nums[right] * nums[right]
            if l_square > r_square:
                res.append(l_square)
                left += 1
            else:
                res.append(r_square)
                right -= 1
        res.reverse()
        return res